applications/asset_tracker_v2/src/events/location_module_event.h

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef _LOCATION_MODULE_EVENT_H_
+#define _LOCATION_MODULE_EVENT_H_
+
+/**
+ * @brief Location module event
+ * @defgroup location_module_event Location module event
+ * @{
+ */
+
+#include <nrf_modem_gnss.h>
+#include <modem/lte_lc.h>
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+#include <net/nrf_cloud_pgps.h>
```

Comment:
Is it?
`#include <net/nrf_cloud_pgps.h>` is flagged in other files where it's used.

Not sure I understand why guarding or not guarding `nrf_cloud_pgps.h` affects on the CONFIG_NRF_CLOUD_AGPS flagging in this file.

PS. I'm playing with AGPS flagging though but it's not related to this comment.

Reply:
Tried to build with and without the `CONFIG_NRF_CLOUD_PGPS` guards in `location_module_event.h`. Works fine.

Reply:
Yep, obviously we can remove those from this file and everything works as we are just adding an unused member into a union. Is that the way we want to put it?
I honestly don't know how we should flag these kind of things as there are so many versions within the code base.

Reply:
Yeah. My opinion is that if it is possible to avoid having guards its best not to include it. Ifdefs really disturbs the flow when looking at the code and can be nightmarish to handle.
Also, if P-GPS functionality is used without the P-GPS config being enabled there should be a build error.
I can imagine how this would look if we were to do this for all header files that include headers for submodules that are optional.
Maybe the other reviewers have some input they want to share on this matter. @balaji-nordic @jtguggedal @jhirsi @tokangas

Reply:
I don't disagree that reading the code with ifdefs is a nightmare. However, if we remove flagging here:
```
#if defined(CONFIG_NRF_CLOUD_PGPS)
		/** Data for event LOCATION_MODULE_EVT_PGPS_NEEDED. */
		struct gps_pgps_request pgps_request;
#endif
```
We won't get compiler error if we try to set this in the code in places which are not PGPS flagged. It may be fine but I'm not sure about that.

For header file, I can probably just remove the flagging although we've flagged especially PGPS pretty accurately in location library because it's a feature which is not on by default.

Reply:
I've removed flagging from includes but not from the union. Add a comment and reopen this if this is not fine.

Reply:
Unit tests didn't pass without `CONFIG_NRF_CLOUD_PGPS` flagging as I don't have nrf_cloud_pgps.h available there. So I put ifdefs back. I guess I have actually added the one into the union because of this originally.